### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by ayaankhan98**

### CODE REVIEW

File: sorting/quick_sort.cpp [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+ *      correct position in sorted array and put all smaller elements (samller
+ *      than x) before x, and put all greater elements (greater than x) after
+ *      x. All this should be done in linear time
+ *
  */
 
 #include <cstdlib>
 #include <iostream>
 
 int partition(int arr[], int low, int high) {
```
Comment (kvedala) [inline sorting/quick_sort.cpp:29]: can you add a brief comment to the individual functions as well.

**Comment by kvedala** [comment]
👍 
the comment blocks should be like this
```c
/** <- start of comment block 


      end of comment block 
 */
```

**Comment by ayaankhan98** [comment]
i think all seems to be fine now

**Comment by kvedala** [comment]
yes, 👍

**Comment by ayaankhan98** [comment]
finally merging this

File: sorting/quick_sort.cpp [hunk 2]
```diff
@@ -37,14 +69,14 @@ void quickSort(int arr[], int low, int high) {
...
+// prints the array after sorting
 void show(int arr[], int size) {
     for (int i = 0; i < size; i++)
         std::cout << arr[i] << " ";
     std::cout << "\n";
 }
 
 // Driver program to test above functions
-
 int main() {
```
Comment (kvedala) [inline sorting/quick_sort.cpp:80]: @ayaankhan98 
There is a dynamic memory allocation for the array `int *arr` on line 84 which should be freed before the program termination.

Add the line just before the return of the main function.
```c
delete [] arr;
```

Comment (ayaankhan98) [reply sorting/quick_sort.cpp:80]: #803

**Pull request merged**